lms/static/js/learner_dashboard/views/entitlement_unenrollment_view.js

```diff
@@ -0,0 +1,128 @@
+                        this.$el.css('position', 'fixed');
+                    }
+                },
+
+                handleSubmit: function() {
+                    var apiEndpoint = this.$submitButton.data('entitlementApiEndpoint');
+
+                    if (apiEndpoint === undefined) {
+                        this.setError(gettext('Error: cannot process unenrollment request.'));
+                        return;
+                    }
+
+                    $.ajax({
+                        url: apiEndpoint,
+                        method: 'DELETE',
+                        complete: this.onComplete.bind(this)
+                    });
+                },
+
+                resetModal: function() {
+                    this.$submitButton.removeData();
+                    this.$submitButton.prop('disabled', false);
+                    this.$headerText.empty();
+                    this.$errorText.removeClass('entitlement-unenrollment-modal-error-text-visible');
+                    this.$errorText.empty();
+                },
+
+                setError: function(message) {
+                    this.$submitButton.prop('disabled', true);
+                    this.$errorText.empty();
+                    HtmlUtils.setHtml(
+                        this.$errorText,
+                        message
+                    );
+                    this.$errorText.addClass('entitlement-unenrollment-modal-error-text-visible');
+                },
+
+                setHeaderText: function(courseName, courseNumber) {
+                    this.$headerText.empty();
+                    HtmlUtils.setHtml(
+                        this.$headerText,
+                        HtmlUtils.interpolateHtml(
+                            gettext('Are you sure you want to unenroll from {courseName} ({courseNumber})? You will be refunded the amount you paid.'),
+                            {
+                                courseName: courseName,
+                                courseNumber: courseNumber
+                            }
+                        )
+                    );
+                },
+
+                setSubmitData: function(apiEndpoint) {
+                    this.$submitButton.removeData();
+                    this.$submitButton.data('entitlementApiEndpoint', apiEndpoint);
+                },
+
+                onComplete: function(xhr) {
+                    var status = xhr.status,
+                        message = xhr.responseJSON && xhr.responseJSON.detail;
+
+                    if (status === 204) {
+                        window.location.href = this.dashboardPath;
+                    } else if (status === 401 && message === 'Authentication credentials were not provided.') {
+                        window.location.href = this.signInPath + '?next=' + encodeURIComponent(this.dashboardPath);
+                    } else {
+                        this.setError(gettext('Error: something went wrong while processing your request.'));
```

Comment:
@catong Could you please confirm that this text is acceptable to display in the entitlement unenrollment modal when something goes wrong processing the unenrollment request.

Reply:
@amangano-edx is this error used only for this specific case, or is it an error we have to keep more generic because it's used in other places/cases?

Reply:
This is a generic error.

Reply:
My suggestion: "Your request could not be processed. Try again later."

Reply:
Changed.